service/amqp/src/test/java/org/kaazing/gateway/service/amqp/amqp091/AmqpOpenCloseHandshakeIT.java

```diff
@@ -0,0 +1,129 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.service.amqp.amqp091;
+
+import static org.kaazing.test.util.ITUtil.createRuleChain;
+
+import java.security.Principal;
+import java.util.Map;
+
+import javax.security.auth.Subject;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.login.LoginException;
+import javax.security.auth.spi.LoginModule;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.kaazing.gateway.server.test.GatewayRule;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.k3po.junit.annotation.ScriptProperty;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+public class AmqpOpenCloseHandshakeIT {
+
+
+    private K3poRule k3po = new K3poRule().setScriptRoot("org/kaazing");
+
+    private GatewayRule gateway = new GatewayRule() {
+        {
+            // @formatter:off
+            GatewayConfiguration configuration = new GatewayConfigurationBuilder()
+                    .service()
+                    .accept("wsn://localhost:8333/amqp")
+                    .connect("tcp://localhost:8334")
+                    .type("amqp.proxy")
+                    .property("service.domain","localhost")
+                    .property("encryption.key.alias", "session")
+                    .realmName("demo")
+                    .authorization()
+                        .requireRole("AUTHORIZED")
+                    .done()
+                .done()
+                .security()
+                    .realm()
+                        .name("demo")
+                        .description("Kaazing WebSocket Gateway Demo")
+                        .httpChallengeScheme("Basic")
+                        .loginModule()
+                            .type("class:" + BasicLoginModule.class.getName())
+                            .success("requisite")
+                        .done()
+                    .done()
+                .done()
+            .done();
+            // @formatter:on
+            init(configuration);
+        }
+    };
+
+    @Rule
+    public TestRule chain = createRuleChain(gateway, k3po);
+    
+    
+    @Test
+    @ScriptProperty({ "connectLocation \"http://localhost:8333/amqp\"", "acceptLocation \"tcp://localhost:8334\"" })
+    @Specification({ "specification/amqp/ws/ws.connect", "specification/amqp/ws/open/identity/request", "specification/amqp/ws/close/request",  
+        "specification/amqp/tcp/open/tcp.accept", "specification/amqp/tcp/open/identity/response", "specification/amqp/tcp/close/response"})
+    public void openAndCloseHandshake() throws Exception {
```

Comment:
I would expect that you would have a method here for each method in the k3po AmqpIndenityTcpIT (or at least two methods, one for openWithIdentityAndClose and one for openWithoutIdentityAndClose)